From: Hannes Reinecke <hare@suse.de>
Date: Tue, 16 Aug 2016 13:56:23 +0200
Subject: scsi_dh_alua: Retry RTPG on SCSI host_bytes values
References: bsc#993866
Patch-Mainline: submitted to linux-scsi, 2016/08/18

When the command is failed with host_bytes values DID_SOFT_ERROR,
DID_IMM_RETRY, or DID_REQUEUE it should be retried directly.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index b60fb6c..e773b73 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -549,6 +549,10 @@ static int alua_rtpg(struct scsi_device *sdev, struct alua_port_group *pg)
 				    "%s: rtpg failed, result %d\n",
 				    ALUA_DH_NAME, retval);
 			kfree(buff);
+			if (host_byte(retval) == DID_SOFT_ERROR ||
+			    host_byte(retval) == DID_IMM_RETRY ||
+			    host_byte(retval) == DID_REQUEUE)
+				goto retry;
 			if (driver_byte(retval) == DRIVER_ERROR)
 				return SCSI_DH_DEV_TEMP_BUSY;
 			return SCSI_DH_IO;
-- 
1.8.5.6

